### PR TITLE
Added examples for specifying options for Grinder

### PR DESCRIPTION
**Pull request by SAY-droid427**
Fixes https://github.com/AdoptOpenJDK/openjdk-tests/issues/2541

### CODE REVIEW

**Comment by SAY-droid427** [comment]
@lxia Could you please tell me what are the examples needed to be added?
NOTE: Even though the pull request shows 4 commits, the changes included in this pull request are only those for the testJobTemplate and not the Wildfly tests.

**Comment by llxia** [comment]
The [Screenshots in the issue](https://github.com/adoptium/aqa-tests/issues/2541) has a great example. Could you include it?
Also, please rebase your branch and squash the git commits into one.

**Comment by SAY-droid427** [comment]
@lxia Please review the latest changes. I will rebase and squash it after all the required changes have been made. Thank you!

**Comment by SAY-droid427** [comment]
@llxia Please review the changes

File: buildenv/jenkins/testJobTemplate [hunk 1]
```diff
@@ -326,8 +326,14 @@ ARCH_OS_LIST.each { ARCH_OS ->
 							    sectionHeaderStyle(sectionHeaderHelpTextStyleCss)
 							}
 							choiceParam('TEST_FLAG', ['', 'JITAAS', 'AOT'], "Optional. Only set to JITAAS/AOT for JITAAS/AOT feature testing.")
-							stringParam('EXTRA_OPTIONS', EXTRA_OPTIONS, "Use this to append options to the test command")
-							stringParam('JVM_OPTIONS', "", "Use this to replace the test original command line options")
+							stringParam('EXTRA_OPTIONS', EXTRA_OPTIONS, "Use this to append options to the commandline",'''
+							Example :<br>
+							\'''-Xjit:{*LoaderThread.run{}V}{very-hot}\(traceCG,log=methodLog\)'\'''<br>
```
Comment (llxia) [inline buildenv/jenkins/testJobTemplate:331]: I do not think `'''` is correct. https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/482/
I tried double quotes and it did not work. And I also tried the example in  https://github.com/adoptium/aqa-systemtest/issues/188. @Mesbah-Alam any idea why it does not work? https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/480/

**Comment by llxia** [comment]
Depends on the underline test framework, the value may be different. `-Xjit:{java/util/HashMap.*}(optlevel=noopt)` works just fine in openjdk tests, but it does not work in system tests.
https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/479/

**Comment by SAY-droid427** [comment]
@llxia Do I need to make any changes?

**Comment by llxia** [comment]
@SAY-droid427 no, not yet. 
Re https://github.com/adoptium/aqa-tests/pull/2549#discussion_r624061959, @Mesbah-Alam could you provide an example? Thanks

**Comment by Mesbah-Alam** [comment]
We can document the following two JIT options as examples of how JIT options can be passed in via `EXTRA_OPTIONS` parameter of grinders to be consumed by STF based system tests: 

```
Extra_Options='-Xjit:count=0,\{java/lang/reflect/Method.getParameterAnnotations*\}\(traceFull,log=getParameterAnnotations.log\)'
```

(Note: the one above worked after braces were escaped).

The [second one](https://github.com/adoptium/aqa-systemtest/issues/188) is what Lan mentioned above. This one worked by using double quotes with escape character (\"..\"), e.g: 

```
-Xjit:\"{java/util/HashMap.*}(optlevel=noopt)\"
```

**Comment by llxia** [comment]
@SAY-droid427 could you update the PR based on https://github.com/adoptium/aqa-tests/pull/2549#issuecomment-840812951? Thanks

**Comment by SAY-droid427** [comment]
@llxia I have updated the PR. Please review. Thanks!

File: buildenv/jenkins/testJobTemplate [hunk 2]
```diff
@@ -326,8 +326,16 @@ ARCH_OS_LIST.each { ARCH_OS ->
...
 							}
 							choiceParam('TEST_FLAG', ['', 'JITAAS', 'AOT'], "Optional. Only set to JITAAS/AOT for JITAAS/AOT feature testing.")
-							stringParam('EXTRA_OPTIONS', EXTRA_OPTIONS, "Use this to append options to the test command")
-							stringParam('JVM_OPTIONS', "", "Use this to replace the test original command line options")
+							stringParam('EXTRA_OPTIONS', EXTRA_OPTIONS, "Use this to append options to the commandline",'''
+							Example 1 :<br>
+							Extra_Options='-Xjit:count=0,\{java/lang/reflect/Method.getParameterAnnotations*\}\(traceFull,log=getParameterAnnotations.log\)'
+							Example 2:<br>
+							-Xjit:\"{java/util/HashMap.*}(optlevel=noopt)\"
+							''')
```
Comment (llxia) [inline buildenv/jenkins/testJobTemplate:334]: stringParam takes three parameters:
`stringParam(String parameterName, String defaultValue = null, String description = null)`

It is incorrect to add an additional parameter. The example should be a part of the description.
Also, the examples are specific to system tests, we should clearly state this.


```
Use this to append options to the commandline. For system tests, the special characters may need to be escaped in EXTRA_OPTIONS and JVM_OPTIONS. 
Example 1:
'-Xjit:count=0,\{java/lang/reflect/Method.getParameterAnnotations*\}\(traceFull,log=getParameterAnnotations.log\)'
Example 2:
-Xjit:\"{java/util/HashMap.*}(optlevel=noopt)\"
```

We can use the same example for  `JVM_OPTIONS`.

Comment (SAY-droid427) [reply buildenv/jenkins/testJobTemplate:335]: Thank you @lxia. I didn't know this format. I have corrected the mistake. Please review. If there are any more mistakes I will correct them immediately.

File: buildenv/jenkins/testJobTemplate [hunk 3]
```diff
@@ -326,8 +326,20 @@ ARCH_OS_LIST.each { ARCH_OS ->
...
-							stringParam('JVM_OPTIONS', "", "Use this to replace the test original command line options")
+							stringParam('EXTRA_OPTIONS', EXTRA_OPTIONS,'''
+							Use this to append options to the commandline. In general, JVM options can be directly used. For system tests, the special characters may need to be escaped in EXTRA_OPTIONS and JVM_OPTIONS.<br> 
+                            Example 1:<br>
+                            '-Xjit:count=0,\{java/lang/reflect/Method.getParameterAnnotations*\}\(traceFull,log=getParameterAnnotations.log\)'
+                            Example 2:<br>
+                            -Xjit:\"{java/util/HashMap.*}(optlevel=noopt)\"
+							''')
+							stringParam('JVM_OPTIONS', "", '''
+							Use this to append options to the commandline. In general, JVM options can be directly used. For system tests, the special characters may need to be escaped in EXTRA_OPTIONS and JVM_OPTIONS.<br>
```
Comment (llxia) [inline buildenv/jenkins/testJobTemplate:337]: For JVM_OPTIONS, please keep the original description as the first sentence. That is, change `Use this to append options to the commandline` -> `Use this to replace the test original command line options. In general, JVM options can be directly used...`. The rest description and example can be the same as EXTRA_OPTIONS

Comment (SAY-droid427) [reply buildenv/jenkins/testJobTemplate]: Done. Once you approve the change I will squash the commits.

**Review by llxia — APPROVED** [review]
Thanks @SAY-droid427

**Review by Mesbah-Alam — APPROVED** [review]
LGTM

**Pull request merged**